interfaces/prompting/prompting.go

```diff
@@ -25,6 +25,17 @@ import (
 	"time"
 )
 
+// Metadata stores information about the origin or applicability of a prompt or
+// rule.
+type Metadata struct {
+	// User is the UID of the subject (user) triggering the applicable requests.
+	User uint32
```

Comment:
Nitpick on the type, let's decide if it's an int or an uint32 and add a type definition for it. The code will read cleaner and we will have fewer places to fix in case we got the guess wrong.